### PR TITLE
Add watchdog parameter-validation, and fix up misleading comments

### LINKED ISSUE #1238: watchdog_enable() confuses milliseconds and microseconds
In the file:

https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_watchdog/include/hardware/watchdog.h

delay_ms is described as:

"delay_ms Number of milliseconds before watchdog will reboot without watchdog_update being called. Maximum of 0x7fffff, which is approximately 8.3 seconds"

But 0x7fffff is 8388607 which is 8.3 seconds only if the unit is microseconds. The text also mentions "then the delay_ms parameter will not be in microseconds".

So I guess it is microseconds? Then I also think the parameter should be named delay_us, not delay_ms.

### PR DESCRIPTION
**Pull request by lurch**
Fixes #1238 and #1550 (and #1241 #1409 #1505 which are duplicates of #1238)

### CODE REVIEW

File: src/rp2_common/hardware_watchdog/include/hardware/watchdog.h [hunk 1]
```diff
@@ -72,7 +77,7 @@ void watchdog_update(void);
  * onto the RPI-RP2), then this value will be cleared, and so \ref watchdog_enable_caused_reboot will
  * return false.
  *
- * \param delay_ms Number of milliseconds before watchdog will reboot without watchdog_update being called. Maximum of 0x7fffff, which is approximately 8.3 seconds
+ * \param delay_ms Number of milliseconds before watchdog will reboot without watchdog_update being called. Maximum of 8388, which is approximately 8.3 seconds
```
Comment (JamesH65) [inline src/rp2_common/hardware_watchdog/include/hardware/watchdog.h:80]: 8.3 IIF the source clock is default?

File: src/rp2_common/hardware_watchdog/include/hardware/watchdog.h [hunk 2]
```diff
@@ -30,17 +30,22 @@
...
  * \note If \ref watchdog_start_tick value does not give a 1MHz clock to the watchdog system, then the \p delay_ms
- * parameter will not be in microseconds. See the datasheet for more details.
+ * parameter will not be in milliseconds. See the datasheet for more details.
  *
  * By default the SDK assumes a 12MHz XOSC and sets the \ref watchdog_start_tick appropriately.
  *
  * \param pc If Zero, a standard boot will be performed, if non-zero this is the program counter to jump to on reset.
  * \param sp If \p pc is non-zero, this will be the stack pointer used.
- * \param delay_ms Initial load value. Maximum value 0x7fffff, approximately 8.3s.
+ * \param delay_ms Initial load value. Maximum value 8388, approximately 8.3s.
```
Comment (JamesH65) [inline src/rp2_common/hardware_watchdog/include/hardware/watchdog.h:48]: 8.3 IIF the source lock is default?

Comment (lurch) [reply src/rp2_common/hardware_watchdog/include/hardware/watchdog.h:48]: Well, you could argues that the `\note If \ref watchdog_start_tick value does not give a 1MHz clock to the watchdog system, then the \p delay_ms  parameter will not be in milliseconds. See the datasheet for more details.` already covers that ? :man_shrugging:

File: src/rp2_common/hardware_watchdog/watchdog.c [hunk 3]
```diff
@@ -34,6 +35,7 @@ uint32_t watchdog_get_count(void) {
 // tag::watchdog_enable[]
 // Helper function used by both watchdog_enable and watchdog_reboot
 void _watchdog_enable(uint32_t delay_ms, bool pause_on_debug) {
+    valid_params_if(WATCHDOG, delay_ms <= 8388); // i.e. floor(0xffffff / 2000)
```
[the remark this answers is not in the document]
Comment (lurch) [reply src/rp2_common/hardware_watchdog/watchdog.c:38]: _Could_ do, but I'm not sure if it's worth it when it only appears once in a debug-level assert?

**Review by kilograham — APPROVED** [review]

**Pull request merged**